core/src/main/java/org/stellar/anchor/util/SepLanguageHelper.java

```diff
@@ -0,0 +1,105 @@
+package org.stellar.anchor.util;
+
+import static org.stellar.anchor.util.Log.errorEx;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.stellar.anchor.api.exception.SepValidationException;
+import org.stellar.anchor.config.AppConfig;
+
+public class SepLanguageHelper {
+  static Map<String, Iso3316Language> fallbackLangs = null;
+
+  public static String validateLanguage(AppConfig appConfig, String lang) {
+    if (fallbackLangs == null) {
+      fallbackLangs = prepareFallbackLanguages(appConfig);
+    }
+
+    if (lang != null) {
+      List<String> languages = appConfig.getLanguages();
+      if (languages != null && languages.size() > 0) {
+        if (languages.stream().noneMatch(l -> l.equalsIgnoreCase(lang))) {
+          return getFallbackLanguage(lang);
+        }
+      }
+      return lang;
+    } else {
+      return getFallbackLanguage(lang);
+    }
+  }
+
+  public static void reset() {
+    fallbackLangs = null;
+  }
+
+  static Map<String, Iso3316Language> prepareFallbackLanguages(AppConfig appConfig) {
+    Map<String, Iso3316Language> fallbacks = new HashMap<>();
+    appConfig
+        .getLanguages()
+        .forEach(
+            lang -> {
+              try {
+                Iso3316Language isoLang = Iso3316Language.of(lang);
+                fallbacks.putIfAbsent(isoLang.getLangKey(), isoLang);
+              } catch (SepValidationException ex) {
+                errorEx(ex);
+              }
+            });
+    return fallbacks;
+  }
+
+  static final String LANGUAGE_ONLY_DEFAULT = "en";
+  static final String LANGUAGE_DEFAULT = "en-US";
+
+  static String getFallbackLanguage(String lang) {
+    try {
+      Iso3316Language language = Iso3316Language.of(lang);
+      Iso3316Language fallback = fallbackLangs.get(language.getLangKey());
+      if (fallback == null) {
+        if (language.getLocale() == null) {
+          return LANGUAGE_ONLY_DEFAULT;
+        }
+        return LANGUAGE_DEFAULT;
+      }
+      return fallback.toString();
+    } catch (SepValidationException e) {
+      return LANGUAGE_DEFAULT;
+    }
+  }
+}
+
+@Getter
+@AllArgsConstructor
+class Iso3316Language {
```

Comment:
Can you leave a link to this ISO? I didn't find it in my search...

The representation I found (similar to what you showed here) was a combination of [ISO-639-1](https://en.wikipedia.org/wiki/ISO_639-1) and  [ISO-3166-1](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2): https://www.fincher.org/Utilities/CountryLanguageList.shtml

Reply:
Thanks for bringing this up. I checked SEP-24 again. The `lang` field is RFC4646. 
https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0024.md#request

However, SEP-6 uses ISO-369-1. 
https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0006.md#request

I think we should make the SEPs consistent. 

I had renamed the class to RFC4646Language and left a link.